File: java/socketd/src/main/java/org/noear/socketd/transport/stream/SubscribeStream.java
package org.noear.socketd.transport.stream;

import org.noear.socketd.transport.core.Reply;
import org.noear.socketd.utils.IoConsumer;

/**
 * 订阅流
 *
 * @author noear
 * @since 2.3
 */
public interface SubscribeStream extends Stream<SubscribeStream> {
    /**
     * 答复发生时
     */
    SubscribeStream thenReply(IoConsumer<Reply> onReply);
}

File: swift/socketd/socketd/src/socketd/transport/stream/SubscribeStream.swift
//
//  SubscribeStream.swift
//  socketd
//
//  Created by noear on 2024/1/12.
//

import Foundation

/**
 * 订阅流
 *
 * @author noear
 * @since 2.3
 */
protocol SubscribeStream : Stream{
    /**
     * 答复发生时
     */
    func  thenReply(_ onReply:IoConsumer<Reply>) -> Self;
    
}
